executor/main_test.go

```diff
@@ -15,13 +15,18 @@
 package executor
 
 import (
-	"os"
 	"testing"
 
 	"github.com/pingcap/tidb/util/testbridge"
+	"go.uber.org/goleak"
 )
 
 func TestMain(m *testing.M) {
 	testbridge.WorkaroundGoCheckFlags()
-	os.Exit(m.Run())
+	opts := []goleak.Option{
+		goleak.IgnoreTopFunction("go.etcd.io/etcd/pkg/logutil.(*MergeLogger).outputLoop"),
+		goleak.IgnoreTopFunction("go.opencensus.io/stats/view.(*worker).start"),
+		goleak.IgnoreTopFunction("gopkg.in/natefinch/lumberjack%2ev2.(*Logger).millRun"),
```

Comment:
but still I don't know where this function comes from.

Reply:
IIRC, `outputLoop` comes from the etcd client. I don't know where `lumberjack` is. You may try ` go mod graph`.
`

Reply:
@xhebox I found it comes from `pingcap/log`

```go
func initFileLog(cfg *FileLogConfig) (*lumberjack.Logger, error) {
  return &lumberjack.Logger
}
```

`millRun` seems like a daemon goroutine that range on `millCh` which is never close.

```go
func (l *Logger) millRun() {
	for _ = range l.millCh {
		// what am I going to do, log this?
		_ = l.millRunOnce()
	}
}
```

Reply:
https://github.com/natefinch/lumberjack/issues/56, It's a goroutine leak bug.